frost/bip340.go

```diff
@@ -61,18 +83,22 @@ func (b *Bip340Curve) SerializePoint(p *Point) []byte {
 }
 
 // DeserializePoint deserializes byte slice to an elliptic curve point. The
-// byte slice length must be equal to SerializedPointLength(). Otherwise,
+// byte slice length must be equal to SerializedPointLength(). The deserialized
+// point must be a valid, non-identity point lying on the curve. Otherwise,
 // the function returns nil.
 func (b *Bip340Curve) DeserializePoint(bytes []byte) *Point {
-
-	// TODO: validate if point is on the curve
-
 	x, y := b.Unmarshal(bytes)
 	if x == nil || y == nil {
 		return nil
 	}
 
-	return &Point{x, y}
+	point := &Point{x, y}
+
+	if !b.IsNotIdentity(point) {
```

Comment:
The naming of this function is a bit confusing in context, because we're also checking that it's generally a valid point, not just that it's not the point at infinity. Should we leave a clarifying comment or rename it?

Reply:
Agreed, this naming is unfortunate. Please see ff1b58dc0e90afe4bfd8affc80328e17845e1d77.